src/traces/box/cross_trace_calc.js

```diff
@@ -90,18 +90,33 @@ function setPositionOffset(traceType, gd, boxList, posAxis, pad) {
     var groupgap = fullLayout[traceType + 'groupgap'];
     var padfactor = (1 - gap) * (1 - groupgap) * dPos / fullLayout[numKey];
 
+    // Find maximum trace width
+    // we baseline this at dPos
+    var max_half_width = dPos;
+    for(i = 0; i < boxList.length; i++) {
+        calcTrace = calcdata[boxList[i]];
+
+        if(calcTrace[0].trace.width) {
+            if(calcTrace[0].trace.width / 2 > max_half_width) {
```

Comment:
Hmm. What if `trace.width ` is smaller than the minimum-diff-between-distinct-values? Does this still work?

Reply:
It actually does work. See [this line](https://github.com/plotly/plotly.js/blob/eb5e9da1d476351244faff9b69cc9c53d53b9238/src/traces/box/cross_trace_calc.js#L118) where the width is set.

These lines are affecting the axis autoscale of the plotted points. But if a violin has trace.width = 0.00001 (much less than the min-diff as you mention) then that trace will shrink to the whatever the width is set to but the "camera" will not zoom in anymore. The view of all the traces is always calibrated to the violin/box with the maximum width set, which doesn't loose any desired layout that you could want with violins or boxes, if you decide to use width at all.